realm-annotations-processor/src/test/resources/io/realm/SimpleRealmProxy.java

```diff
@@ -77,23 +79,23 @@ public static void validateTable(ImplicitTransaction transaction) {
         if(transaction.hasTable("class_Simple")) {
             Table table = transaction.getTable("class_Simple");
             if(table.getColumnCount() != 2) {
-                throw new IllegalStateException("Column count does not match");
+                throw new RealmMigrationNeededException("Column count does not match");
```

Comment:
You change to "field" in the messages below. Maybe it should be "field" here too.